### ISSUE TITLE
Programs spawned with 'exec' pass arguments to exec instead of the program

### ISSUE THREAD

**Issue by kubouch** [nushell/nushell#2675]
**Describe the bug**
In some cases, the `exec` command (which seems to be evoked internally) does not recognize arguments passed to the spawned program. Apparently, it thinks the arguments belong to `exec` and throws an error.

**To Reproduce**
**Reproducible ONLY when nushell is a default login shell.** When nu is launched from other default shell (e.g., bash) the issue does not appear.

I encountered this issue specifically under two circumstances:
1. Using customized ssh commands from `~/.ssh/config`
2. Passing arguments to `gdb --args`

**Steps to reproduce the behavior:**

0. Set nushell as the default login shell, log out and log back in.

1. Put a ProxyCommand config into `~/.ssh/config`, for example:
```
Host ServerUseX
    ProxyCommand ssh -XC user@server.xyz
```
2. `ssh ServerUseX` 
```
error: exec unexpected -XC
  ┌─ shell:1:12
  │
1 │ exec ssh -XC user@server.xyz
  │            ^ unexpected flag (try exec -h)
```
3. `ssh -XC user@server.xyz`
... works fine
4. `gdb --args find ~ -name config.toml`, type 'run', hit enter
```
error: exec unexpected -name
  ┌─ shell:1:38
  │
1 │ exec /usr/bin/find /home/user -name config.toml
  │                                   ^ unexpected flag (try exec -h)

```

**Expected behavior**
I expected the commands to run normally.

**Configuration (please complete the following information):**
 - OS (e.g. Windows): *Arch Linux, 5.8.14-arch1-1*
 - Nu version (you can use the `version` command to find out): *0.21.0, 2df8775b485817d6bd528e938d062a52bdda5bc3*
 - Optional features (if any): *default, clipboard trash*

**Comment by kubouch** [comment]
Something seems to be wrong with exec. When I type `exec ls` or `exec 'ps aux'` (the second is from exec --help message), the whole nu crashes.

**Comment by fdncred** [comment]
@kubouch I asked that question of @thegedge a while back. See it here [in the PR](https://github.com/nushell/nushell/pull/2495#issuecomment-687602374).

**Comment by kubouch** [comment]
OK so closing nu after exec is intended. This issue still persists though. I'm pretty sure it's just nushell passing the arguments to exec instead of the spawned process.

**Comment by jakevossen5** [comment]
I can confirm that this also is an issue macOS 11.2.2

**Comment by kubouch** [comment]
OK, I looked into this and a partial solution is to wrap flag-like arguments into quotes, like this (excerpt from `~/.ssh/config`):
```shell
Host ServerUseX
    ProxyCommand ssh '-XC' user@server.xyz
```
instead of
```shell
Host ServerUseX
    ProxyCommand ssh -XC user@server.xyz
```

Doesn't seem to work with gdb, though -- maybe it internally stripts the quotes or something like that.

The core of the problem is that the programs spawn the commands with `exec`, assuming that every shell == POSIX shell. For example, when you run `ssh ServerUseX`, ssh spawns the following:
```shell
exec ssh -XC user@server.xyz
```
but then it finds nushell's `exec` which tries to interpret `-XC` as its flag.

I think the best solution would be to add an option to tell the parser: "Just give me all args as strings and don't bother parsing them". I'm not sure yet whether it would be doable just by implementing a new SyntaxShape or by hard-coding it into the parser, I haven't looked at the parser code yet. @jonathandturner, what do you think about this? Perhaps I could take this as my first leap into the nushell's internals.

**Comment by fdncred** [comment]
I noticed some weird quote handling with external the other day too related to #3363. I wonder if these things are related.
https://github.com/nushell/nushell/blob/a8f555856a65412a664327b9b6cbee5ec753e0b2/crates/nu-command/src/commands/classified/external.rs#L104-L122

**Comment by kubouch** [comment]
> I noticed some weird quote handling with external the other day too related to #3363. I wonder if these things are related.
> 
> https://github.com/nushell/nushell/blob/a8f555856a65412a664327b9b6cbee5ec753e0b2/crates/nu-command/src/commands/classified/external.rs#L104-L122

Might be related, I wasn't even aware that nushell supports escapes. I tried escapes with the gdb command but it made some random output -- so the issue you linked might be related to that.

I see the quotes as a workaround, though. Imagine if you don't have the option to modify the ssh proxy command, then you'd be screwed. I was thinking nushell could adopt some very very limited subset of POSIX behavior that some applications depend on, like exec in this case, to make life easier in certain specialized cases.

**Comment by fdncred** [comment]
Nushell doesn't support escapes but since these are external commands, external commands support escapes, typically. That's my rationale.

**Comment by kubouch** [comment]
I don't have the issue on 0.33 anymore so I'll close it and we can reopen if somebody runs into it again.

**Comment by kubouch** [comment]
Seems like people are stumbling on the "exec" issue from time to time. It works with the workaround wrapping the flag in quotes but sometimes this is not an option when using some 3rd party tool.

**Comment by RAOF** [comment]
> Seems like people are stumbling on the "exec" issue from time to time. It works with the workaround wrapping the flag in quotes but sometimes this is not an option when using some 3rd party tool.

Particularly, there doesn't seem to be any way of getting `gdb --args` to call `exec` in a way that nushell is happy with, which is somewhat of a pain :)

**Comment by kubouch** [comment]
@RAOF I managed to work around gdb by setting the `SHELL` environment variable to bash. In Nushell, you can do this, for example: `with-env { SHELL: /bin/bash } { gdb --args program args... }`

**Comment by dzintars** [comment]
Encountered this as well.
Solved via:
```ini
Host hp
  # ProxyJump jump.host.com
  ProxyCommand ssh '-W' '[target.host.com]:22' jump.host.com
```

**Comment by fdncred** [comment]
closing as fixed by #7527. please re-open if it's not fixed.
